lib/interaction_machine.js

```diff
@@ -670,6 +672,7 @@ var InteractionMachine = Eventable.extend(function(self, api, app) {
                 return self.state.continue_session();
             })
             .then(function(continue_session) {
+                opts.helper_metadata = self.state.helper_metadata;
```

Comment:
Going with the utils.functor approach partially for consistency and partially to cater for unexpected use-cases (e.g. needing to retrieve the name of a voice file via an http request).